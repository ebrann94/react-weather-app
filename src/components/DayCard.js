import React from 'react';

const DayCard = (props) => (
    <div className="day-card">
        <h4>{props.day}</h4>
        <p>{props.temperature} &deg;C</p>
        <p>{props.windSpeed}mph {props.windDirection}</p>
    </div>
);

DayCard.defaultProps ={
    day: '- - -',
    temperature: '-',
    windSpeed: '-',
    windDirection: '-'
}

export default DayCard;