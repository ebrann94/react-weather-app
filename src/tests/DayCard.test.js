import React from 'react';
import { shallow } from 'enzyme';
import DayCard from '../components/DayCard';

describe('DayCard component tests', () => {
    it('renders with props', () => {
        const props = {
            day: 'Monday',
            img: '/img1',
            windSpeed: 2,
            windDirection: 'N',
            temperature: 12
        }

        const wrapper = shallow(<DayCard {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
})