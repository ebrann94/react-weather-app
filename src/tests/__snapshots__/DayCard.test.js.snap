// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayCard component tests renders with props 1`] = `ShallowWrapper {}`;
